util.py

```diff
@@ -438,3 +438,10 @@ def getPlainURL(self):
             assert self.password is None
 
             return self.url
+
+    def getBaseURL(self):
+        """Get the URL with schema and network location without username/password."""
+
+        url = self.getPlainURL()
+        parts = urllib.parse.urlsplit(url)
+        return urllib.parse.urlunsplit(parts._replace(path='', query='', fragment=''))
```

Comment:
I think it would be better do something like this, rather than relying on a private, semi-undocumented method:

```
return urllib.parse.urlunsplit(urllib.parse.SplitResult(parts.scheme, parts.netloc, '', '', ''))
```

Reply:
done